Makefile

```diff
@@ -3,7 +3,12 @@ REPO_ROOT                                      := $(shell dirname $(realpath $(l
 GITHUBORG                                      ?= open-component-model
 OCMREPO                                        ?= ghcr.io/$(GITHUBORG)/ocm
 VERSION                                        := $(shell go run api/version/generate/release_generate.go print-rc-version $(CANDIDATE))
-EFFECTIVE_VERSION                              := $(VERSION)+$(shell git rev-parse HEAD)
+COMMIT                                         = $(shell git rev-parse --verify HEAD)
+# if EFFECTIVE_VERSION is not set, set it to VERSION+HEAD
+# this is not the same as '?=' because it will also set the value if EFFECTIVE_VERSION is set to an empty string
+ifeq ($(EFFECTIVE_VERSION),)
+EFFECTIVE_VERSION                              := $(VERSION)+$(COMMIT)
+endif
 GIT_TREE_STATE                                 := $(shell [ -z "$$(git status --porcelain 2>/dev/null)" ] && echo clean || echo dirty)
 COMMIT                                         := $(shell git rev-parse --verify HEAD)
```

Comment:
what's the difference between: `:=` and `=`?
Since `COMMIT` is now already defined in line 6, we don't need it in line 13, do we?

```suggestion
```